### PR TITLE
GOVUKAPP-1029: Search results header

### PR DESCRIPTION
**Pull request by davidc-gds**
# Title

Add header to search results and focus on it when search results are displayed

## JIRA ticket(s)
  - [GOVUKAPP-1029](https://govukverify.atlassian.net/browse/GOVUKAPP-1029)

## Figma
  - [Figma board](https://www.figma.com/design/5u5cvVCgKY3Rng2ADNV30K/2024-07---GOV.UK-app---beta-designs?node-id=3600-27558&t=KNLJY4udtFS7ebEx-4)

## Screen shots & video (if UI changes)
Screenshot_20241217_145114

[Screen_recording_20241217_145145.webm](https://github.com/user-attachments/assets/64e3364b-e61e-4ac2-9101-6fa620e2d27c)

[GOVUKAPP-1029]: https://govukverify.atlassian.net/browse/GOVUKAPP-1029?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: feature/search/src/main/kotlin/uk/govuk/app/search/ui/SearchScreen.kt [hunk 1]
```diff
@@ -124,63 +133,90 @@ private fun SearchScreen(
...
-                        GovUkTheme.spacing.medium,
-                        GovUkTheme.spacing.medium,
-                        0.dp
-                    ),
-                    onClick = {
-                        onClick(title, url)
-                        context.startActivity(intent)
-                    }
+        item {
+            Title3BoldLabel(
```
Comment (TimGJD) [inline feature/search/src/main/kotlin/uk/govuk/app/search/ui/SearchScreen.kt:151]: Judging by this PR https://github.com/alphagov/govuk-mobile-android-app/pull/155 we might want to check if this should be set as a semantic header as I think it fits the criteria?

File: feature/search/src/main/kotlin/uk/govuk/app/search/ui/SearchScreen.kt [hunk 2]
```diff
@@ -124,63 +133,90 @@ private fun SearchScreen(
 }
 
 @Composable
-fun ShowResults(searchResults: List<Result>, onClick: (String, String) -> Unit) {
-    Column(
-        modifier = Modifier
-            .padding(bottom = GovUkTheme.spacing.medium)
-            .fillMaxSize(),
-        horizontalAlignment = Alignment.CenterHorizontally
+fun ShowResults(
```
Comment (TimGJD) [inline feature/search/src/main/kotlin/uk/govuk/app/search/ui/SearchScreen.kt:136]: Can this be private whilst we're here?

File: feature/search/src/main/kotlin/uk/govuk/app/search/ui/SearchScreen.kt (hunk 1 above)
Comment (davidc-gds) [reply feature/search/src/main/kotlin/uk/govuk/app/search/ui/SearchScreen.kt:151]: Good shout!

Comment (davidc-gds) [reply feature/search/src/main/kotlin/uk/govuk/app/search/ui/SearchScreen.kt:153]: Done

**Pull request merged**